From ff67288d481aa1ec15e8f70a4a60796d655160b8 Mon Sep 17 00:00:00 2001
From: Marc Mutz <marc.mutz@qt.io>
Date: Wed, 20 Mar 2024 10:41:56 +0100
Subject: [PATCH 106/130] QObjectPrivate::Signal: initialize all members

A recent change in that area triggered Clang-SA to rescan the code and
complain about this pre-existing bug: If receiver == nullptr, then the
`previous` member was never initialized.

Fix by null'ing it using NSDMI. This is a trivial type, the compiler
will be able to avoid the redundant write.

Amends ab92b9e40025dcf08c14232de762a268201a78b4(!).

Pick-to: 6.7 6.6 6.5 6.2 5.15
Change-Id: Ideed71f0f36d5f896fb6a4614f233757c1371ee3
Reviewed-by: David Faure <david.faure@kdab.com>
Reviewed-by: Edward Welbourne <edward.welbourne@qt.io>
(cherry picked from commit 40714c1ddd886f6d0bde53e05f27d6b3e9e6649b)
---
 src/corelib/kernel/qobject_p.h | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/corelib/kernel/qobject_p.h b/src/corelib/kernel/qobject_p.h
index 13ffb88999d..325a974f308 100644
--- a/src/corelib/kernel/qobject_p.h
+++ b/src/corelib/kernel/qobject_p.h
@@ -220,7 +220,7 @@ public:
                 s = s->previous;
             }
         }
-        Sender *previous;
+        Sender *previous = nullptr;
         QObject *receiver;
         QObject *sender;
         int signal;
-- 
2.39.5 (Apple Git-154)

